src/game/resource.c

```diff
@@ -1,27 +1,80 @@
 #include "resource.h"
 
 #include "building/type.h"
+#include "core/image.h"
+#include "core/image_group_editor.h"
+#include "game/save_version.h"
 #include "scenario/building.h"
+#include "translation/translation.h"
 
-int resource_image_offset(resource_type resource, resource_image_type type)
-{
-    if (resource == RESOURCE_MEAT && scenario_building_allowed(BUILDING_WHARF)) {
-        switch (type) {
-            case RESOURCE_IMAGE_STORAGE: return 40;
-            case RESOURCE_IMAGE_CART: return 648;
-            case RESOURCE_IMAGE_FOOD_CART: return 8;
-            case RESOURCE_IMAGE_ICON: return 11;
-            default: return 0;
-        }
-    } else {
-        return 0;
+#define RESOURCE_ALL RESOURCE_MAX + RESOURCE_SPECIAL
+
+static const resource_type resource_mappings[][RESOURCE_ALL] = {
+    {
+        RESOURCE_NONE, RESOURCE_WHEAT, RESOURCE_VEGETABLES, RESOURCE_FRUIT, RESOURCE_OLIVES, RESOURCE_VINES,
```

Comment:
That is the original resource order, which I haven't changed yet.

My idea is, as soon as this PR gets merged, we reorder the resources to make more sense. In fact I already have that code running locally.

The first resource mapping will always have this order, though. As the name implies, this is what maps the old resource ID to the new resource ID.

The idea is that you can change the ID for each resource enum. Today `RESOURCE_WHEAT` is `1`, but tomorrow it can be changed to `2`. By having this array pointing index `1` to `RESOURCE_WHEAT`, it will always point to the current id of that enum even if you change it.